substrate/runtime/example/src/lib.rs

```diff
@@ -0,0 +1,385 @@
+	// to attach a bond to the operation. as the first major alteration of storage, reserve
+	// some value from the sender's account (`Balances` module has a `reserve` function for
+	// exactly this scenario). This amount should be enough to cover any costs of the
+	// substantial execution in case it turns out that you can't proceed with the operation.
+	//
+	// If it eventually transpires that the operation is fine and, therefore, that the
+	// expense of the checks should be borne by the network, then you can refund the reserved
+	// deposit. If, however, the operation turns out to be invalid and the computation is
+	// wasted, then you can burn it or repatriate elsewhere.
+	//
+	// Security bonds ensure that attackers can't game it by ensuring that anyone interacting
+	// with the system either progresses it or pays for the trouble of faffing around with
+	// no progress.
+	//
+	// If you don't respect these rules, it is likely that your chain will be attackable.
+	fn accumulate_dummy(_aux: &T::PublicAux, increase_by: T::Balance) -> Result {
+		// Read the value of dummy from storage.
+		let dummy = Self::dummy();
+		// Will also work using the `::get` on the storage item type iself:
+		// let dummy = <Dummy<T>>::get();
+
+		// Calculate the new value.
+		let new_dummy = dummy.map_or(increase_by, |dummy| dummy + increase_by);
+
+		// Put the new value into storage.
+		<Dummy<T>>::put(new_dummy);
+		// Will also work with a reference:
+		// <Dummy<T>>::put(&new_dummy);
+
+		// Let's deposit an event to let the outside world this happened.
+		Self::deposit_event(RawEvent::Dummy(increase_by));
+
+		// All good.
+		Ok(())
+	}
+
+	// Implementation of a priviledged call. This doesn't have an `aux` parameter because
+	// it's not (directly) from an extrinsic, but rather the system as a whole has decided
+	// to execute it. Different runtimes have different reasons for allow priviledged
+	// calls to be executed - we don't need to care why. Because it's priviledged, we can
+	// assume it's a one-off operation and substantial processing/storage/memort can be used
+	// without worrying about gamability or attack scenarios.
+	fn set_dummy(new_value: T::Balance) -> Result {
+		// Put the new value into storage.
+		<Dummy<T>>::put(new_value);
+
+		// All good.
+		Ok(())
+	}
+}
+
+// The trait expresses what should happen when the block is finalised.
+impl<T: Trait> OnFinalise<T::BlockNumber> for Module<T> {
+	fn on_finalise(_: T::BlockNumber) {
+		// Anything that needs to be done at the end of the block.
+		// We just kill our dummy storage item.
+		<Dummy<T>>::kill();
+	}
+}
+
+#[cfg(feature = "std")]
+#[derive(Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+#[serde(deny_unknown_fields)]
+/// The genesis block configuration type. This is a simple default-capable struct that
+/// contains any fields with which this module can be configured at genesis time.
```

Comment:
For my education, what is "genesis time" here? Is it the initial state of the contract? So unrelated to the "block chain genesis" right?

Reply:
The initial state of the blockchain, which dictates this module's initial storage items' values.